command/agent/config/config.go

```diff
@@ -123,6 +126,11 @@ func LoadConfig(path string) (*Config, error) {
 		return nil, errwrap.Wrapf("error parsing 'cache':{{err}}", err)
 	}
 
+	err = parseTemplates(&result, list)
+	if err != nil {
```

Comment:
Nit: If this is instead phrased like this:
```
if err := parseTemplates(&result, list); err != nil {
    ...
}
```
It's generally considered more defensive because the error can't be reused by another function below (`err = somethingElse`). That's considered more defensive because it protects against weird bugs where someone forgets to check an error, and then a failure happens from it being populated _earlier_, and it protects against races in case somebody starts using the err in a goroutine further down below.

Reply:
I agree, I was following the pattern of `parseListeners` and `parseCache` calls above this (indeed, I probably just copied one of those and changed the method). 

In this situation the error cannot be reused because if `err` is not nil, we immediately return.

I went ahead and changed all three invocations of this style to the `if err:=;err!=nil` style in 09072a172